app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,50 @@
+class ReviewsController < ApplicationController
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @review = Review.new(review_params)
+    @beer = Beer.find(params[:beer_id])
+    if @review.save
+      flash[:notice] = "Review Saved"
+      redirect_to beer_path(@beer)
+    else
+      flash[:notice] = "Invalid Input"
+      render :new
+    end
+  end
+
+  def edit
+    @review = Review.find(params[:id])
+  end
+
+  def update
+    @review = Review.find(params[:id])
+    @beer = Beer.find(params[:beer_id])
+    if @review.update(review_params)
+      flash[:notice] = "Review Updated"
+      redirect_to beer_path(@beer)
+    else
+      flash[:notice] = "Review not updated"
+      render :edit
+    end
+  end
+
+  def destroy
+    @beer = Beer.find(params[:beer_id])
+    @review = Review.find(params[:id])
+    if @review.destroy
+      flash[:notice] = "Review deleted"
+    else
+      flash[:notice] = "Review not deleted"
+    end
+    redirect_to beer_path(@beer)
+  end
+
+  protected
+
+  def review_params
+    params.require(:review).permit(:rating, :description, :beer_id, :user_id)
```

Comment:
Don't accept `:user_id` through the params. This would allow a user to impersonate someone else just by changing their ID to some arbitrary value.

The `beer_id` is passed via the URL (`/beers/10/reviews`) so you can extract the ID there rather than through the form.